python/clicastInterface.py

```diff
@@ -348,18 +350,76 @@ def executeTest(enviroPath, testIDObject):
     return executeReturnCode, stdoutText
 
 
-def generateExecutionReport(enviroPath, testIDObject):
+def generate_report(enviroPath, testObject):
+    """
+    Generates the our custom report for the test case execution data
+
+    File gets written to output
+    """
+
+    # Calculate the location of our custom folder
+    source_root = pathlib.Path(__file__).parent.resolve()
+    custom_dir = source_root / "custom"
+
+    # What's the path to our custom CSS?
+    custom_css = custom_dir / "vscode.css"
+
+    # Patch get_option to use our CSS without setting the CFG option
+    monkeypatch_custom_css(custom_css)
 
-    standardArgs = getStandardArgsFromTestObject(testIDObject, False)
-    # We build a clicast command script to generate the execution report
-    # since we need multiple commands
+    # Set in the config that the format we use is HTML
     with open(commandFileName, "w") as commandFile:
-        # we force report mode to HTML just to be safe
         commandFile.write("option VCAST_CUSTOM_REPORT_FORMAT HTML\n")
-        commandFile.write(
-            standardArgs + " report custom actual " + testIDObject.reportName
-        )
 
-    # we ignore the exit code and return the stdout
-    exitCode, stdOutput = runClicastScript(enviroPath, commandFileName)
-    return stdOutput
+    runClicastScript(enviroPath, commandFileName)
+
+    # Open-up the unit test API
+    with UnitTestApi(testObject.enviroName) as api:
+        # Find and check for our unit
+        unit_found = False
```

Comment:
I'd remove this; I think we have to "trust" that the arguments passed in will always work, rather than checking like this.

Reply:
Removed in b47303ebd7cd9ac952b379fb59fcb96cdRemoved in b47303ebd7cd9ac952b379fb59fcb96cd0b4c57d0b4c57d

Reply:
I reduced the check logic... I somehow find it "uncomfortable" to not throw anything if we would not find the test case or if again some "special" string like "not-used" is added in the future.